tool/plugin/lib/plugin.dart

```diff
@@ -657,20 +753,57 @@ class DeployCommand extends ProductCommand {
 /// Generate the plugin.xml from the plugin.xml.template file.
 /// If the --release argument is given, create a git branch and
 /// commit the new file to it, assuming the release checks pass.
-class GenCommand extends Command {
+/// Note: The product-matrix.json file includes a build spec for the
+/// EAP version at the end. When the EAP version is released that needs
+/// to be updated.
+class GenCommand extends ProductCommand {
   final BuildCommandRunner runner;
 
   GenCommand(this.runner);
 
   String get description =>
-      'Generate a valid plugin.xml and .travis.yaml for the Flutter plugin.\n'
+      'Generate a valid plugin.xml and .travis.yml for the Flutter plugin.\n'
       'The plugin.xml.template and product-matrix.json are used as input.';
 
   String get name => 'gen';
 
-  Future<int> run() async {
-    // TODO(messick): Implement GenCommand.
-    throw 'unimplemented';
+  Future<int> doit() async {
+    List json = readProductMatrix();
+    var spec = new SyntheticBuildSpec.fromJson(json.first, release, json.last);
+    log('writing pluginl.xml');
+    int value = 1;
+    var result = await genPluginXml(spec, 'resources');
+    if (result != null) {
+      log('writing .travis.yml');
+      genTravisYml(specs);
+      value = 0;
+    }
+    if (isReleaseMode) {
+      await createGitBranch(release);
+    }
+    return new Future(() => value);
+  }
+
+  Future<int> createGitBranch(String relNumber) async {
```

Comment:
Having the script create git branches may limit what we can do in terms of fixing issues in branches or running the build release multiple times. Perhaps this could instead enforce constraints - that `build --release` must be run on a branch w/ the right name pattern?